src/plugins/intel_gpu/src/kernel_selector/cl_kernels/permute_f_y_axes.cl

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/fetch_data.cl"
+
+#define READ_VEC(offset, ptr)          CAT(vload, VEC_SIZE)(offset, ptr)
+#define WRITE_VEC(val, offset, ptr)    CAT(vstore, VEC_SIZE)(val, offset, ptr)
+
+#define IN_VEC_TYPE                     MAKE_VECTOR_TYPE(INPUT0_TYPE, VEC_SIZE)
+#define TO_IN_VEC_TYPE(x)               CAT(convert_, IN_VEC_TYPE)(x)
+#define ACC_VEC_TYPE                    MAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, VEC_SIZE)
+#define TO_ACC_VEC_TYPE(x)              CAT(convert_, ACC_VEC_TYPE)(x)
+#define OUT_VEC_TYPE                    MAKE_VECTOR_TYPE(OUTPUT_TYPE, VEC_SIZE)
+#define TO_OUT_VEC_TYPE(x)              CAT(convert_, OUT_VEC_TYPE)(x)
+
+#if defined (PERMUTE_SIMPLE_MEM_COPY)
+KERNEL (permute_f_y_axes)(
+    const __global INPUT0_TYPE* input,
+    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+    , FUSED_OPS_DECLS
+#endif
+    )
+{
+    const int bf = get_global_id(2);
+    const int f_idx = bf % INPUT0_FEATURE_NUM;
+    const int b_idx = bf / INPUT0_FEATURE_NUM;
+    const int x_start = get_global_id(0) * BLOCK_SIZE;
+    const int y_idx = get_global_id(1);
+
+    __attribute__((opencl_unroll_hint(J_TIMES)))
+    for (int j = 0; j < J_TIMES; ++j) {
+        const int x_idx = x_start + j * VEC_SIZE;
+        IN_VEC_TYPE res = READ_VEC(0, &input[INPUT0_GET_INDEX(b_idx, f_idx, y_idx, x_idx)]);
+#if HAS_FUSED_OPS
+        FUSED_OPS;
+        OUT_VEC_TYPE result  = FUSED_OPS_RESULT;
+#else
+        OUT_VEC_TYPE result = ACTIVATION(res, ACTIVATION_PARAMS);
+#endif
+        const int output_idx = OUTPUT_GET_INDEX(b_idx, y_idx, f_idx, x_idx);
+        WRITE_VEC(result, 0, &output[output_idx]);
+    }
+}
+
+#elif defined (THREE_DIM_TRANSPOSE)
+
```

Comment:
I am curious why not using subgroup read/ write as my suggestion before.

Reply:
Will fix this, by the way, do you mean to use vload/vstore or BLOCK_READ macro that use subgroup extension?

Reply:
Current impl is not I suggested. I suggested to use "subgoup read" https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroups.html
so that each work items will read /write contiguous block in the block direction. 
Please contact me if you need further explanation. My initial suggestion for blocked format is quite simple to implement.